homeassistant/components/mysensors/gateway.py

```diff
@@ -162,9 +164,11 @@ async def _get_gateway(
         persistence_file = hass.config.path(persistence_file)
 
     if device == MQTT_COMPONENT:
-        # what is the purpose of this?
-        # if not await async_setup_component(hass, MQTT_COMPONENT, entry):
-        #    return None
+        # make sure the mqtt integration is set up
+        if not await async_setup_component(
```

Comment:
Let's use `async_when_setup` instead. You can import it from `homeassistant.setup`. And continue setup when it's done.

Reply:
Sure, but if the MQTT integration can't be set up, calling `async_setup_component` is going to fail too. So isn't that the same result, except that we don't have to start setting up integrations in other integrations, but can leave that to HA instead.

Reply:
My point is that we want the direct feedback we get from `async_setup_component` when it fails.

Reply:
Since MQTT Is part of the `after_dependencies`, the fact that it's not in `hass.config.components` would be enough of an indication?

Reply:
Yes, you're right. That should be enough. 👍